src/Validation/Constraints/Custom.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/* Copyright (c) 2017 Stefan Hecken <stefan.hecken@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\Validation\Constraints;
+use ILIAS\Validation\Constraint;
+use ILIAS\Data\Factory;
+use ILIAS\Data\Result;
+
+class Custom implements Constraint {
+	/**
+	 * @var ILIAS\Data\Factory
+	 */
+	protected $data_factory;
+
+	/**
+	 * @var callable
+	 */
+	protected $is_ok;
+
+	/**
+	 * @var callable
+	 */
+	protected $error;
+
+	/**
+	 * @param string|callable   $error
+	 */
+	public function __construct(callable $is_ok, $error, Factory $data_factory) {
```

Comment:
For readabilities sake it would help me if this was `Data\Factory` instead of `Factory`.